compiler/jsgen.nim

```diff
@@ -2043,7 +2043,11 @@ proc genMagic(p: PProc, n: PNode, r: var TCompRes) =
   of mNew, mNewFinalize: genNew(p, n)
   of mChr: gen(p, n[1], r)
   of mArrToSeq:
+    # initializing typed arrays doesn't need copy
     if needsNoCopy(p, n[1]):
+      # we change the kind from tyArray to tySequence because 
+      # `genArrayConstr` needs it.
+      skipTypes(n[1].typ, abstractVarRange).kind = tySequence
```

Comment:
this sounds fishy for several reasons:
* it modifies source `n` (and doesn't reset it)
* what if it's inside some `compiles` statement, would this side effect affect code?
* looks like it violates how case statement discriminant fields should be set

instead, can't we somehow pass a flag to `gen` so it handles it appropriately?

Reply:
Yeah, it seems not to be case statement discriminant fields